File: javascript/leetcode/852. Peak Index in a Mountain Array.js
/**
 * @param {number[]} A
 * @return {number}
 */
var peakIndexInMountainArray = function (A) {
  const ans = 0;

  for (let i = A.length - 2; i >= 0; --i) if (A[i] < A[i + 1]) return ++i;

  return ans;
};

File: c++/leetcode/852. Peak Index in a Mountain Array.cpp
class Solution
{
public:
  int peakIndexInMountainArray(vector<int> &A)
  {
    int ans = 0;

    for (int i = A.size() - 2; i >= 0; --i)
    {
      if (A[i] < A[i + 1])
        return ++i;
    }

    return ans;
  }
};